serverless/serverless
# Remove hard-coded AWS partitions

Also removing unused method that had hard-coded partition inside

## What did you implement

- Removed some hard-coded aws partitions to account for non-standard partitions such as `aws-cn` or [`aws-us-gov`](https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/using-govcloud-arns.html)
- Removed unused method that included a hard-coded partition


## How can we verify it

Tricky to verify since would need an AWS account than can select gov-cloud or China regions, but deploy a function with ALB as event source in a gov-cloud or China region.

```yaml
service: my-gov-cloud-app

provider:
  name: aws
  runtime: python3.6
  region: us-gov-west-1  # gov-cloud region

functions:
  albEventConsumer:
    handler: handler.hello
    events:
      - alb:
          # notice the partition in the ARN
          listenerArn: arn:aws-us-gov:elasticloadbalancing:us-east-1:12345:listener/app/my-load-balancer/50dc6c495c0c9188/
          priority: 1
          conditions:
            path: /hello
  myLambdaAtEdge:
    handler: myLambdaAtEdge.handler
    events:
      - cloudFront:  # not currently possible in gov-cloud, but may be soon
          eventType: viewer-response
          origin: s3://bucketname.s3.us-gov-west-1.amazonaws.com/files
```

Note Cloudfront isn't available for govcloud regions, but is available in [China](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services/)

## Todos

<details>
<summary>Useful Scripts</summary>


- `npm run test:ci` --> Run all validation checks on proposed changes
- `npm run lint:updated` --> Lint all the updated files
- `npm run lint:fix` --> Automatically fix lint problems (if possible)
- `npm run prettier-check:updated` --> Check if updated files adhere to Prettier config
- `npm run prettify:updated` --> Prettify all the updated files

</details>


**_Is this ready for review?:_** YES
**_Is it a breaking change?:_** NO
